### ISSUE TITLE
`fn` that returns a future and contains a single `async` block should be warned to use `async fn` instead

### ISSUE THREAD

**Issue by khuey** [rust-lang/rust#71391]
Something like

```rust
fn foo(args) -> impl Future<Output = u32> {
  async move {
    // Do stuff with args.
    my_u32
  }
}
```

`async fn` is strictly better.

**Comment by jonas-schievink** [comment]
Sounds like a good fit for a Clippy lint

**Comment by Mark-Simulacrum** [comment]
Yeah, I think this is a poor fit for rustc itself for now, so I'm going to go ahead and close -- it seems likely that it could have false positives (in particular such a writing style might be preferred by some people, especially when refactoring).
